iFixit/react-commerce
# Upgrade algolia instantsearch

closes #1382 
closes #1487 

This PR introduces the new `react-instantsearch-hooks-router-nextjs` package.
With this new package, the nextjs router now plays nicely with Algolia.

On the Algolia side there is still [this pending issue](https://github.com/algolia/instantsearch/issues/5571) that has been mitigated by a little hack

### QA

1. Visit Vercel preview
2. Verify that the browser back button is working correctly
3. Verify that the item type name persisting in the hero section issue is now fixed
